pkg/security/probe/probe.go

```diff
@@ -329,8 +343,66 @@ func (p *Probe) DispatchEvent(event *model.Event) {
 		handler.HandleEvent(event)
 	}
 
-	// Process after evaluation because some monitors need the DentryResolver to have been called first.
-	p.monitor.ProcessEvent(event)
+	// TODO: split this func
+	if p.Config.SecurityProfileAnomalyDetection && event.ProfileState == model.MatchedAndAbsent && event.GetEventType() != model.SyscallsEventType {
+		p.monitor.securityProfileManager.FillProfileContextFromContainerID(event.ProcessContext.ContainerID, &event.SecurityProfileContext)
+		if event.GetEventType() == model.FileOpenEventType {
+			if p.Config.SecurityProfileFilesBestEffort {
+				fmt.Printf("FILE Event not found in profile -> generate anomaly detection ! %s @ %+v for %s\n",
+					event.ProcessContext.Comm, event.GetEventType().String(), event.Open.File.PathnameStr)
+				if p.anomalyDetectionRateLimiterOpen.Allow() {
+					p.DispatchCustomEvent(
+						events.NewCustomRule(events.AnomalyDetectionRuleID),
+						events.NewCustomEvent(model.AnomalyDetectionOpenEventType, serializers.NewEventSerializer(event, p.resolvers)),
+					)
+				} else {
+					fmt.Printf("  -> Event rate limited\n")
+				}
+			}
+		} else if event.GetEventType() == model.DNSEventType {
+			fmt.Printf("DNS Event not found in profile -> generate anomaly detection ! %s @ %+v for %s (%d)\n",
+				event.ProcessContext.Comm, event.GetEventType().String(), event.DNS.Name, event.DNS.Type)
+			if p.anomalyDetectionRateLimiterDNS.Allow() {
+				p.DispatchCustomEvent(
+					events.NewCustomRule(events.AnomalyDetectionRuleID),
+					events.NewCustomEvent(model.AnomalyDetectionDNSEventType, serializers.NewEventSerializer(event, p.resolvers)),
+				)
+			} else {
+				fmt.Printf("  -> Event rate limited\n")
+			}
+		} else if event.GetEventType() == model.BindEventType {
+			fmt.Printf("BIND Event not found in profile -> generate anomaly detection ! %s @ %+v for %s\n",
+				event.ProcessContext.Comm, event.GetEventType().String(), event.DNS.Name)
+			if p.anomalyDetectionRateLimiterBind.Allow() {
+				p.DispatchCustomEvent(
+					events.NewCustomRule(events.AnomalyDetectionRuleID),
+					events.NewCustomEvent(model.AnomalyDetectionBindEventType, serializers.NewEventSerializer(event, p.resolvers)),
+				)
+			} else {
+				fmt.Printf("  -> Event rate limited\n")
+			}
+		} else if event.GetEventType() == model.ExecEventType {
+			fmt.Printf("EXEC Event not found in profile -> generate anomaly detection ! %s @ %+v\n",
+				event.ProcessContext.Comm, event.GetEventType().String())
+			if p.anomalyDetectionRateLimiterProcess.Allow() {
```

Comment:
any reason to not use the rate limiters defined here : https://github.com/DataDog/datadog-agent/blob/main/pkg/security/module/rate_limiter.go#L33